photutils/detection/tests/test_peakfinder.py

```diff
@@ -118,28 +113,25 @@ def test_wcs_values(self):
     def test_constant_array(self):
         """Test for empty output table when data is constant."""
         data = np.ones((10, 10))
-        with catch_warnings(NoDetectionsWarning) as warning_lines:
+        with pytest.warns(NoDetectionsWarning, match='Input data is constant'):
             tbl = find_peaks(data, 0.)
             assert tbl is None
-            assert len(warning_lines) > 0
-            assert ('Input data is constant.' in
-                    str(warning_lines[0].message))
 
     def test_no_peaks(self):
         """
         Test for an empty output table with the expected column names
         when no peaks are found.
         """
-        with catch_warnings(NoDetectionsWarning):
+        with pytest.warns(NoDetectionsWarning):
             tbl = find_peaks(IMAGE, 10000)
             assert tbl is None
-
+        with pytest.warns(NoDetectionsWarning):
```

Comment:
I think checking for the warning in every call that it is expected from is more correct, hence the extra lines. Otherwise, you will not know if one of them suddenly stops emitting the warning if they are all dumped into the same context manager.